optimum/intel/neural_compressor/trainer.py

```diff
@@ -59,40 +60,93 @@
 
 __version__ = "4.22.2"
 
+
 logger = logging.get_logger(__name__)
 
 
-class IncTrainer(Trainer):
-    def train(
+from itertools import chain
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Union
+
+from transformers.data.data_collator import DataCollator
+from transformers.modeling_utils import PreTrainedModel
+from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+from transformers.trainer_callback import TrainerCallback
+from transformers.trainer_utils import EvalPrediction
+from transformers.training_args import TrainingArguments
+
+from neural_compressor import training
+from neural_compressor.conf.pythonic_config import _BaseQuantizationConfig
+from neural_compressor.model.torch_model import PyTorchModel
+from optimum.exporters import TasksManager
+
+
+class INCTrainer(Trainer):
+    """
+    INCTrainer enables Intel Neural Compression quantization aware training, pruning and distillation.
+    """
+
+    def __init__(
         self,
-        agent: Optional[Component] = None,
-        resume_from_checkpoint: Optional[Union[str, bool]] = None,
-        trial: Union["optuna.Trial", Dict[str, Any]] = None,
-        ignore_keys_for_eval: Optional[List[str]] = None,
-        **kwargs,
+        model: Union[PreTrainedModel, torch.nn.Module] = None,
+        args: TrainingArguments = None,
+        data_collator: Optional[DataCollator] = None,
+        train_dataset: Optional[Dataset] = None,
+        eval_dataset: Optional[Dataset] = None,
+        tokenizer: Optional[PreTrainedTokenizerBase] = None,
+        model_init: Callable[[], PreTrainedModel] = None,
+        compute_metrics: Optional[Callable[[EvalPrediction], Dict]] = None,
+        callbacks: Optional[List[TrainerCallback]] = None,
+        optimizers: Tuple[torch.optim.Optimizer, torch.optim.lr_scheduler.LambdaLR] = (None, None),
+        preprocess_logits_for_metrics: Callable[[torch.Tensor, torch.Tensor], torch.Tensor] = None,
+        quantization_config: Optional[_BaseQuantizationConfig] = None,

```

Comment:
shall we list these configs here as three input parameters or a single parameter which is a list of configs? whether the latter has better extensibility?

Reply:
IMO three parameters make more sense right now as it's more understandable for users but happy to discuss about different possibilities ! Why do you think we should make something more extensible ? Will we have more configuration to pass in the future ? I think ideally we would have one parameter combining all the details related to all the different compression techniques, and create the adapted INC configuration directly in the `INCTrainer`, what do you think ? In case we plan to do this, I think it would be better to add it in a future PR as this one might be already pretty big